src/utils/concatenate_sentence.py

```diff
@@ -0,0 +1,166 @@
+from transformers import pipeline
+
+
+def back_translate(
+    origin_target_data: list,
+    model: str,
+    sample_percent: float,
+) -> List[str]:
+    """
+    Function to back translate a list of sentences using a model from huggingface.
+
+    Args:
+        origin_target_data (list): Data to be back translated.
+        model (str): Model to use for back translation.
+        sample_percent (float): percent of data to back_tranlate.
+
+    Returns:
+        List[str]: List of back translated sentences.
+    """
+    number_of_samples = math.ceil(sample_percent * len(origin_target_data))
+    target_sample_indexes = random.sample(
+        range(len(origin_target_data)), number_of_samples
+    )
+    target_samples = [origin_target_data[index] for index in target_sample_indexes]
+    translator = pipeline("text2text-generation", model=model)
+    translated_source_data = [result["generated_text"] for result in translator(target_samples)]
+    return translated_source_data, target_sample_indexes
+
+
+def concantenate_sentences(
+    subset_name: str,
+    seperator_token: str = "[SEP]",
+    sentence_length_threshold: int = 25,
+    sequential: bool = False,
+    do_back_translation: bool = True,
+    number_of_concatenations: int = 100,
+    back_translation_model: str = None,
+    back_translation_percent: float = None
+) -> Tuple[List[str], List[str]]:
+    """
+    Function to concantenate sentences from a source and target file as described
+    by Kondo et al in Sentence Concatenation Approach to Data Augmentation for Neural.
+
+    Args:
+        subset_name (str): Name of Mafand subset.
+        seperator_token (str, optional): Token to use as separator token. Defaults to
+        "[SEP]".
+        sentence_length_threshold (int, optional): Threshold required for length of
+        sentences from the pseudo data to make it to the final dataset. Defaults to 25.
+        sequential (bool, optional): Concatenates the sentences sequentially if True
+        else it does it randomly. Defaults to False.
+        do_back_translation (bool, optional): Whether to back-translate or not. Defaults
+        to True.
+        number_of_concatenations (int, optional): Number of concatenation to do, this
+        parameter is only needed if `sequential` is False, if `do_back_translation` 
+        is True. Defaults to 100.
+        back_translation_model (str, optional): name of model to use for back translations.
+        This parameter is only needed if `do_back_translation` is True. Defaults to None.
+        back_translation_percent (float, optional): percent of target data to back translate.
+        Should take any value between 0 and 1. This parameter is only needed if 
+        `do_back_translation` is True. Defaults to None.
+
+    Returns:
+        Tuple[List[str], List[str]]: Returns of a tuple of the source and target sentences
+    """
+    # load data
+    data = load_dataset("masakhane/mafand", subset_name)
```

Comment:
There should be an argument that takes the dataset name as an input variable for the script. The default can be `mafand`, but we want the option to set another dataset name. You can use `argparse`.

Reply:
on it